core/src/cluster_info.rs

```diff
@@ -1796,15 +1858,21 @@ impl ClusterInfo {
         requests: Vec<PullData>,
         stakes: &HashMap<Pubkey, u64>,
     ) -> Option<Packets> {
-        // split the requests into addrs and filters
-        let mut caller_and_filters = vec![];
-        let mut addrs = vec![];
         let mut time = Measure::start("handle_pull_requests");
+        self.time_gossip_write_lock("process_pull_reqs", &self.stats.process_pull_requests)
+            .process_pull_requests(requests.iter().map(|r| r.caller.clone()), timestamp());
         self.update_data_budget(stakes);
-        for pull_data in requests {
-            caller_and_filters.push((pull_data.caller, pull_data.filter));
-            addrs.push(pull_data.from_addr);
-        }
+        let mut packets = Packets::new_with_recycler(recycler.clone(), 64, "handle_pull_requests");
+        let (caller_and_filters, addrs): (Vec<_>, Vec<_>) = {
+            let mut rng = rand::thread_rng();
+            let check_pull_request =
+                self.check_pull_request(Instant::now(), &mut rng, &mut packets);
+            requests
+                .into_iter()
+                .filter(check_pull_request)
```

Comment:
It might be helpful for debugging to have datapoint to track the number of requests dropped here. We currently have https://github.com/solana-labs/solana/blob/master/core/src/cluster_info.rs#L1898-L1900 in this function, but this new filter isn't accounted for inside of that

Reply:
yes, that is a good idea. added a new stats counter here.